scout/server/blueprints/public/controllers.py

```diff
@@ -0,0 +1,142 @@
+
+    for case in cases:
+        event_list = events_in_case(store, user, case)
+        events_per_case.append(event_list)
+
+    for event_list in events_per_case:
+        event = {}
+        event["human_readable"] = compile_important_events(event_list)
+        head, *_tail = event_list
+        event["link"] = head["link"]
+        event["case"] = head["case"]
+        events_of_interest.append(event)
+    return events_of_interest
+
+
+def recent_cases(user, store):
+    """Return a list of recent cases order in increasing age. A case may appear only once."""
+    return list(store.unique_cases_by_date({"_id": user.email}))
+
+
+def events_in_case(store, user, case):
+    """Return a list of events associated with a user's specific case"""
+    return list(store.user_events_by_case({"_id": user.email}, case))
+
+
+def compile_important_events(event_list):
+    """Compile a string of recent events, ordered by importance, together with the sum of the events occurences.
+    Importance is determined by placement in the macro VERBS_MAP.
+        Args:
+            event_list
+        Returns:
+            string, example: 'commented on 1 case. Pinned variant X 1.'
+    """
+    pairs = []
+    for event in event_list:
+        pairs.append((event["verb"], event["category"]))
+
+    event_sum_list = sum_occurrences(pairs)
+    LOG.debug(f"EVENTS: {event_sum_list}")
+    top_events = get_important_events(event_sum_list)
+    LOG.debug(f"BEST: {top_events}")
+    return events_to_string(top_events)
+
+
+def get_important_events(all_events):
+    """Sort tuple list accourding to macro VERBS_MAP and return
+    a list of length(NOF_RECENT_EVENTS).
+
+    Args:
+        all_events:Dict()
+
+    Returns:
+        tuple_list: (verb, count)
+    """
+    events_by_importance = sorted(all_events, key=verb_index)
+    i = 0
+    l = []
+    while i < NOF_RECENT_EVENTS and i < len(events_by_importance):
+        key = events_by_importance[i]
+        l.append((key, all_events[key]))
+        i += 1
+    return l
+
+
+def events_to_string(list_of_events):
+    """List of tuples: [(Key, kombo), n_events]
+
+    Returns: string: describing events"""
+    l = []
+
+    def possessive_s(n):
+        if n > 1:
+            return "s"
+        return ""
+
+    def tautology(verb, event):
+        return event in VERBS_MAP.get(verb)
+
+    for event in list_of_events:
+        (verb, event_type), n = event
+        if tautology(verb, event_type):
+            l.append(VERBS_MAP.get(verb) + " X" + str(n))
+        else:
+            l.append(VERBS_MAP.get(verb) + " " + str(n) + " " + event_type + possessive_s(n))
+    return reduce(lambda a, b: a + ", " + b, l)
+
+
+def sum_occurrences(pairs):
+    """Count tuples
+    Args:
+        pairs list of tuples (verb, category)
+
+    Returns:
+        Dict where each key is a tuple(verb, type), the value is numnber of occurances {('pin', 'variant'): 1,
+    """
+
+    def sum_occurrences_aux(pairs, acc):
```

Comment:
Some comments in this function might help understand what the code does. For instance, what's acc, what's tail?

Adding some example mights further help the reader/maintainer

Reply:
Acc is  now spelled out as accumulator.

Tail is a common way to refer to the remaining elements of a linked list.